### PR TITLE
test: categorynavbar component

### PR DESCRIPTION
**Pull request by Diegoalbag**
## Description

Added unit tests for packages/theme/modules/catalog/category/components/navbar/CategoryNavbar.vue
## Screenshots (if appropriate):
Screen Shot 2022-05-05 at 3 25 27 PM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    useUiState.mockReturnValue(uiStateMock);
+    const { getByText } = render(CategoryNavbar, categoryNavbarProps);
+    const button = getByText('Filters');
+
+    await userEvent.click(button);
+
+    expect(uiStateMock.toggleFilterSidebar).toHaveBeenCalledTimes(1);
+  });
+
+  it('Should render "sort by" select and change selection', async() => {
```
Comment (bartoszherba) [inline packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js:30]: Can we also verify that select has all expected options based on its data?

File: packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    useUiState.mockReturnValue(uiStateMock);
+
+    const { getByLabelText } = render(CategoryNavbar, categoryNavbarProps);
+    const gridViewIcon = getByLabelText(/Change to grid view/i);
+
+    await userEvent.click(gridViewIcon);
+
+    expect(uiStateMock.changeToCategoryGridView).toHaveBeenCalledTimes(1);
+  });
+});
```
Comment (bartoszherba) [inline packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js:64]: I think that we can also add the following tests:
1. Check if the items count is rendered properly with the expected value
2. Check if the component is in loading state all skeletons are rendered as expected
3. change sorting action emits 'reloadProducts' event

File: packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import userEvent from '@testing-library/user-event';
+import { render } from '~/test-utils';
+import CategoryNavbar from '../CategoryNavbar.vue';
+import { paginationData, sortByData} from '~/test-utils/mocks/categoryNavbarMock';
+import { expect } from '@jest/globals';
```
Comment (bartoszherba) [inline packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js:5]: Screenshot 2022-05-09 at 09 08 14
It is not required to import assertion anyway.

**Review by bartoszherba — CHANGES_REQUESTED** [review]
@Diegoalbag Please, fix also the PR name so it align with the rules (remove the dot ;) )

File: packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js (hunk 3 above)
Comment (bartoszherba) [reply packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js]: @Diegoalbag What about this one?

File: packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js (hunk 1 above)
Comment (bartoszherba) [reply packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js]: @Diegoalbag What about this one?

File: packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js (hunk 2 above)
Comment (bartoszherba) [reply packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js:65]: @Diegoalbag What about this one?

**Comment by sethidden** [comment]
I recently added a GitHub action step that runs ESLint and fails if there are any ESLint errors.
If the build action starts failing in this PR, please:
1. Rebase the branch from this PR on develop then push out the changes.
2. If that doesn't work, this means your new code has some linting errors. Please go to the repository root (where the .git folder is) and run `yarn lint`

File: packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js (hunk 3 above)
Comment (Diegoalbag) [reply packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js]: @bartoszherba Assertion is not being imported anymore.

File: packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  });
+
+  it('Should render "sort by" select and change selection and select has all expected options based on its data', async() => {
+    const uiHelperMock = useUiHelpersMock();
+    useUiHelpers.mockReturnValue(uiHelperMock);
+
+    const { getByRole, getByText, getAllByRole, emitted } = render(CategoryNavbar, categoryNavbarProps);
+    const sortSelect = getByRole('combobox');
+    const options = getAllByRole('option');
+    expect(options.length).toBe(categoryNavbarProps.props.sortBy.options.length);
```
Comment (Diegoalbag) [inline packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js:36]: @bartoszherba here I verify that select has all expected options based on its data

File: packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    
+    await userEvent.selectOptions(sortSelect, ['name_ASC']);
+    expect(getByText('Sort: Name A-Z').value).toBe(sortSelect.value);
+    expect(emitted().reloadProducts).toHaveProperty('length', 1);
```
Comment (Diegoalbag) [inline packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js:40]: @bartoszherba Here I check that changing sorting action emits 'reloadProducts' event

File: packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  ...jest.requireActual('~/composables'),
+  useUiHelpers: jest.fn(),
+  useUiState: jest.fn(),
+}));
+
+const categoryNavbarProps = { props: { pagination: paginationData, sortBy: sortByData } };
+const loadingCategoryNavbarProps = { props: { pagination: paginationData, sortBy: sortByData, isLoading: true } };
+
+describe('CategoryNavbar.vue', () => {
+  it('Should render Skeleton when loading is true', () => {
```
Comment (Diegoalbag) [inline packages/theme/modules/catalog/category/components/navbar/__tests__/CategoryNavbar.spec.js:19]: @bartoszherba  Check if the component is in loading state all skeletons are rendered as expected

**Review by bartoszherba — APPROVED** [review]

**Pull request merged**